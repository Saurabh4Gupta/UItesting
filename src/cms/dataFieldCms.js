export const dataFieldCms = {
  createNewDataRequest: 'Create new data request',
  editDataRequest: 'Edit data request',
  localMarketLabel: 'Local market',
  localMarketHint: 'Choose the market you wish to create this data request for',
  nameLabel: 'Data request name',
  backLabel:'Back to clients',
  nameHint:
    "Make sure it's clear to help others identify it.Max 255 characters",
  briefLabel: 'Briefing',
  briefHint:
    'Tell us in detail what this data request is for, max 4000 characters',
  templateFileLabel: 'Add tracker template file',
  templateFileHint:
    'Only .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx files in 20mb or less',
  uploadFileLabel:
    'To upload a new file,drag and drop you file below or browse using the link',
  uploadFileLargeMessage: 'File must be less than 20MB',
  labelFileTypeNotAllowed: 'File must be of type .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx',
  uploadButtonText:'Upload new file',
  actualDataLabel: 'Actual data',
  versionHistoryTitle:'Version history',
  actualDataHint: 'Number of months of actual productivity data',
  forecastDataLabel: 'Forecast data',
  forecastDataHint: 'Number of months of forecasted productivity data',
  dueDateLabel: 'Due date',
  dueDateHint: 'Enter the date that you need this data request by',
  assignToLabel: 'Assign to',
  assignToHint: `Only the person or team you assign this to will have access. Once you click 'Create' ,
  they will be sent email notifications asking them to fill in the data request.`,
  create: 'Create',
  upload: 'Upload',
  cancel: 'Cancel',
  productivityDatarequestHeading: 'Productivity data requests',
  productivityDatarequestCaption: ' The ongoing and complete productivity data requests created for this client',
  selectPlaceHolder: 'Please select',
  datePlaceHolder: 'Date',
  viewDetails: 'View Details',
  viewClientsHeading: 'My clients',
  viewClientsCaption: 'Choose the client you wish to view',
  completeLabel: 'Complete',
  ongoingLabel: 'Ongoing',
  emptyProductivityDatarequestCaption:
    'There are currently no active Productivity data requests',
  emptyLabel: 'Create new data request',
  edit: 'Edit',
  actual: 'Actuals',
  forecast: 'Forecast',
  overdue: 'Overdue',
  ongoing: 'Ongoing',
  complete: 'Complete',
  lastUpdate: 'Last Update',
  moveToComplete: 'Move to Complete',
  moveToOnGoing: 'Move to Ongoing',
  delete: 'Delete',
  deletePopConfimationMessage:'Once deleted, this data request will no longer be available.',
  deleteButtonCancelMessage:'No, Cancel',
  deleteButtonDeleteMessage:'Yes, Delete',
  popUpTitle:'Are you sure you want to delete?',
  emptyCompletedProductivityDatarequestCaption:
    'No Productivity data requests have been moved to this section yet.',
  editRequest: 'Edit request',
  save: 'Save',
  productivityRequest: 'Productivity request',
  showMore: 'Show more',
  reportingYear: 'Reporting year',
  productivityDataRequested: 'Productivity data requested',
  actualCapitalized: 'Actual',
  forecastCapitalized: 'Forecast',
  dueBy: 'Due by',
  assignedTo: 'Assigned to',
  trackerTemplate: 'Tracker template',
  close: 'Close',
  reportingYearLabel: 'Reporting year',
  reportingYearHint: 'The reporting year this data request is for',
  toastRequestCreated: 'Data request created',
  toastRequestEdited: 'Data request edited',
  toastRequestDeleted: 'Data request deleted',
  toastFileUploaded: 'File uploaded',
  toastMovedToOngoing: 'Data request moved to ongoing',
  toastMovedToComplete: 'Data request moved to complete',
  currentVersionText:'current version',
  loadMoreText:'Load more',
  uploadFileErrorMessage:'Upload tracker template',
  moveToCompleteConfirmation: 'Are you sure you want to move this to complete?',
  moveToCompleteDescription: 'You will not be abale to make any more edits to this request unless it is moved back to "Ongoing".',
  deletePopUpConfirmation: 'Are you sure you want to delete?',
  deletePopUpDescription: 'Once deleted, this data request will no longer be available.',
  noCancel: 'No, cancel',
  yesContinue: 'Yes, continue',
  yesDelete: 'Yes, delete',
}
