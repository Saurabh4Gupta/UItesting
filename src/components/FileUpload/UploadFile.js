import React, { useState } from 'react';
import { Modal, Button, Dropzone, FormField, Paragraph } from '@dentsu-ui/components';
import PropTypes from 'prop-types';
import constant from '../../utils/constant';

const UploadFile = (props) => {
  const { MAX_FILE_SIZE, ALLOWED_FILE_TYPES } = constant;
  const { cmsData } = props;
  
  const [error, setError] = useState(false);
  const [modalOpen, setModalOpen] = useState(false);

  const [files, setFiles] = useState(
    { files: [] },
  );

  const handleSubmit = (fileLength) => {

    if (!fileLength || fileLength === 0) {
      return setError(true);
      } 
      
     
      setModalOpen(false);
      
    
  };

  const handleCreateData = () => {
    setModalOpen(true);
  };

  const handleInit = () => {
    console.log('Dropzone instance has initialised');
  };

  const onCloseModal = () => {
    setModalOpen(false);
    setFiles({ files: [] });
  };

  return (
    <>
      <Modal
        isFullHeight={false}
        width="540px"
        isOpen={modalOpen}
        onClose={onCloseModal}
      >
        <Modal.Header title="Upload new file" />
        <Modal.Body>
          <FormField
            label={cmsData.uploadFileLabel}
            hint={cmsData.templateFileHint}
          >
            <Dropzone
              allowMultiple={false}
              onInit={() => handleInit()}
              onUpdateFiles={(fileItems) => {
                setFiles(fileItems);
              }}
              maxFiles={1}
              maxFileSize={MAX_FILE_SIZE}
              server="./"
              acceptedFileTypes={ALLOWED_FILE_TYPES}
              dropValidation={true}
            />
          </FormField>
          {error &&  <Paragraph>No file in the Dropzone</Paragraph>}
        </Modal.Body>
        <Modal.Footer>
          <Button
            variant="secondary"
            onClick={onCloseModal}
          >
            cancel
          </Button>
          <Button onClick={() => handleSubmit(files.length)}>Upload</Button>
        </Modal.Footer>
      </Modal>
      <Button bg="rgba(220,220,220,0.4)" icon="upload" onClick={handleCreateData}>
        {cmsData.uploadButtonText}
      </Button>
    </>
  );
};
UploadFile.propTypes = {
  cmsData: PropTypes.object,
}

UploadFile.defaultProps = {
  cmsData: {},
}
export default UploadFile;