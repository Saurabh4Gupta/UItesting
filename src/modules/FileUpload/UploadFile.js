import React, { useState } from 'react';
import { Modal, Button } from '@dentsu-ui/components';
import PropTypes from 'prop-types';
import Toast from '@dentsu-ui/components/dist/cjs/components/Toast';
import Form from '../../components/FileUpload/Form';
import { dataFieldCms as PageContent } from '../../cms';

const UploadFile = (props) => {
  const { cmsData, modalOpen, setModalOpen } = props;
  const [error, setError] = useState({});
  const [files, setFiles] = useState([]);
  const [isDisabled, setDisabled] = useState(true);

  const errorHandler = (errorMsg) => {
    if (errorMsg) {
      setError({ error: `${PageContent.uploadFileErrorMessage}` });
      return true;
    }
    setDisabled(false)
    setError({ error: null });
    return false;
  };

  const validate = () => {
    if (!files.length) {
      setError({ error: `${PageContent.uploadFileErrorMessage}` });
      return false;
    }
    return true;
  };

  const onSubmit = () => {
    const isValid = validate();
    if (isValid) {
      setModalOpen(false);
      if (!error.error && files.length > 0) {
        const toast = new Toast();
        return toast({
          title: '',
          content: PageContent.toastFileUploaded,
          status: 'success',
        });
      }
   }

    return null;
  };

  const onCloseModal = () => {
    setModalOpen(false);
    setError({});
    setFiles([]);
  };

  const handleFileChange = (fileItems) => {
    setDisabled(true)
    setFiles(fileItems);
    setError({ error: null });
    if (files.length > 0) {
      setDisabled(false)
    }
  };
  return (
    <>
      <Modal
        isFullHeight={false}
        width="650px"
        isOpen={modalOpen}
        onClose={onCloseModal}
      >
        <Modal.Header title={PageContent.uploadButtonText} />
        <Modal.Body>
          <Form
            handleFileChange={handleFileChange}
            files={files}
            errors={error}
            cmsData={cmsData}
            setError={errorHandler}
          />
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={onCloseModal}>
            {cmsData.cancel}
          </Button>
          <Button isDisabled={isDisabled} onClick={() => onSubmit()}>{cmsData.upload}</Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};
UploadFile.propTypes = {
  cmsData: PropTypes.object,
  modalOpen: PropTypes.bool,
  setModalOpen: PropTypes.func,
};

UploadFile.defaultProps = {
  cmsData: {},
  modalOpen: false,
  setModalOpen: () => {},
};
export default UploadFile;
