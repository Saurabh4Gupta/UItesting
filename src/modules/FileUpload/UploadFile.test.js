import React from 'react';
import { shallow } from 'enzyme';
import UploadFile from './UploadFile';
import { dataFieldCms as PageContent } from '../../cms';

describe('Success test cases for: UploadFile component', () => {
  let wrapper;

  describe('Snapshot test for  UploadFile component', () => {
    beforeEach(() => {
      wrapper = shallow(<UploadFile cmsData={PageContent} />);
    });

    it('Test Upload click event', () => {
      wrapper.find({ children: PageContent.upload }).simulate('click');
    });
    it('Test Cancel click event', () => {
      wrapper.find({ children: PageContent.cancel }).simulate('click');
    });
    it('Test Cancel click event', () => {
      wrapper.debug().handleFileChange([]);
      console.log('[TESTING]', wrapper.debug())

     // wrapper.find({ children: 'Cancel' }).simulate('click');
    });

   it('invoking handleFileChange functions', () => {
    const pageControls = wrapper.find('ForwardRef').at(2).find('Form');
    expect(pageControls.prop('handleFileChange')());
   })

    it('Test to match the snapsot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Renders correctly', () => {
      expect(wrapper.exists()).toBe(true);
    });
  });
});
