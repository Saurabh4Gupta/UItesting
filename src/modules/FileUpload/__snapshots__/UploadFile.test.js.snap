// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success test cases for: UploadFile component Snapshot test for  UploadFile component Test to match the snapsot 1`] = `
<Fragment>
  <ForwardRef
    isFullHeight={false}
    isOpen={false}
    onClose={[Function]}
    width="650px"
  >
    <ForwardRef
      title="Upload new file"
    />
    <ForwardRef>
      <Form
        cmsData={
          Object {
            "actual": "actuals",
            "actualCapitalized": "Actual",
            "actualDataHint": "Number of months of actual productivity data",
            "actualDataLabel": "Actual Data",
            "assignToHint": "Only the person or team you assign this to will have access. Once you click 'Create' ,
  they will be sent email notifications asking them to fill in the data request.",
            "assignToLabel": "Assign To",
            "assignedTo": "Assigned to",
            "briefHint": "Tell us in detail what this data request is for, max 4000 characters",
            "briefLabel": "Briefing",
            "cancel": "Cancel",
            "close": "Close",
            "complete": "Complete",
            "completeLabel": "Complete",
            "create": "Create",
            "createNewDataRequest": "Create new data request",
            "datePlaceHolder": "Date",
            "delete": "Delete",
            "dueBy": "Due by",
            "dueDateHint": "Enter the date that you need this data request by",
            "dueDateLabel": "Due Date",
            "edit": "Edit",
            "editDataRequest": "Edit data request",
            "editRequest": "Edit request",
            "emptyCompletedProductivityDatarequestCaption": "No Productivity data requests have been moved to this section yet.",
            "emptyLabel": "Create new data request",
            "emptyProductivityDatarequestCaption": "There are currently no active Productivity data requests",
            "forecast": "forecast",
            "forecastCapitalized": "Forecast",
            "forecastDataHint": "Number of months of forecasted productivity data",
            "forecastDataLabel": "Forecast Data",
            "labelFileTypeNotAllowed": "File must be of type .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx",
            "lastUpdate": "Last Update",
            "localMarketHint": "Choose the market you wish to create this data request for",
            "localMarketLabel": "Local Market",
            "moveToComplete": "Move to Complete",
            "moveToOnGoing": "Move to Ongoing",
            "nameHint": "Make sure it's clear to help others identify it.Max 255 characters",
            "nameLabel": "Data request name",
            "ongoing": "Ongoing",
            "ongoingLabel": "Ongoing",
            "overdue": "Overdue",
            "productivityDataRequested": "Productivity data requested",
            "productivityDatarequestCaption": " The ongoing and complete productivity data requests created for this client",
            "productivityDatarequestHeading": "Productivity data requests",
            "productivityRequest": "Productivity Request",
            "reportingYear": "Reporting year",
            "reportingYearHint": "The reporting year this data request is for",
            "reportingYearLabel": "Reporting year",
            "save": "Save",
            "selectPlaceHolder": "Please select",
            "showMore": "Show more",
            "templateFileHint": "Only .xls, .xlsx, .csv, .xlsb, .xlsm, .xlt, .xltm, .xltx files in 20mb or less",
            "templateFileLabel": "Add tracker template file",
            "trackerTemplate": "Tracker template",
            "upload": "Upload",
            "uploadButtonText": "Upload new file",
            "uploadFileLabel": "To upload a new file,drag and drop you file below or browse using the link",
            "uploadFileLargeMessage": "File must be less than 20MB",
            "viewClientsCaption": "Choose the client you wish to view",
            "viewClientsHeading": "My clients",
            "viewDetails": "View Details",
          }
        }
        errors={Object {}}
        files={Array []}
        handleFileChange={[Function]}
      />
    </ForwardRef>
    <ForwardRef>
      <ForwardRef
        onClick={[Function]}
        variant="secondary"
      >
        Cancel
      </ForwardRef>
      <ForwardRef
        onClick={[Function]}
      >
        Upload
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</Fragment>
`;
